pymeasure/instruments/attocube/anc300.py

```diff
@@ -226,18 +233,49 @@ class ANC300Controller(Instrument):
         "getcser", """ Serial number of the controller board """
     )
 
-    def __init__(self, host, axisnames, passwd, query_delay=0.05, **kwargs):
-        kwargs['query_delay'] = query_delay
+    _reg_value = re.compile(r"\w+\s+=\s+(\w+)")
+
+    def __init__(
+        self,
+        adapter=None,
+        name="attocube ANC300 Piezo Controller",
+        axisnames="",
+        passwd="",
+        query_delay=0.05,
+        **kwargs,
+    ):
+        if adapter is None:
+            adapter = self.handle_deprecated_host_arg(kwargs)
+
+        self.query_delay = query_delay
+        self.termination_str = "\r\n"
+
         super().__init__(
-            AttocubeConsoleAdapter(host, 7230, passwd, **kwargs),
-            "attocube ANC300 Piezo Controller",
+            adapter,
+            name,
             includeSCPI=False,
+            read_termination=self.termination_str,
+            write_termination=self.termination_str,
             **kwargs
         )
+
         self._axisnames = axisnames
         for i, axis in enumerate(axisnames):
             setattr(self, axis, Axis(self, i + 1))
 
+        self.wait_for()
+        super().read()  # clear messages sent upon opening the connection
+        # send password and check authorization
+        self.write(passwd)
+        self.wait_for()
+        ret: str = super().read()
```

Comment:
In pymeasure we do not use type hints:
```suggestion
        ret = super().read()
```